Snakefile

```diff
@@ -420,92 +637,1344 @@ rule atlas_experiment_summary:
+    params:
+        organism=get_organism()
+    output:
+        decoexpression="{accession}-{metric}.tsv"
+    shell:
+        """
+        set -e # snakemake on the cluster doesn't stop on error when --keep-going is set
+        exec &> "{log}"
+        source {workflow.basedir}/bin/reprocessing_routines.sh
+
+        geneNameFile=$( get_geneNameFile_given_organism {params.organism}  )
+
+        echo $geneNameFile
+
+        test -s "$geneNameFile" || (  >&2 echo "$0 gene name file not found: $geneNameFile" ; exit 1 )
+        #test -s "$FPKMexpressionsFile" -o -s "$TPMexpressionsFile" || (  >&2 echo "$0 no data files for $e" ; exit 1 )
+
+        decoratedFile={output.decoexpression} 
+
+        # Ammonite REPL & Script-Runner
+
+        amm -s {workflow.basedir}/bin/decorateFile.sc \
+        --geneNameFile "$geneNameFile" \
+        --source {input.expression} \
+        | awk 'NR == 1; NR > 1 {{print $0 | "sort -n"}}' \
+        > $decoratedFile.swp
+
+        decoratedFileLength=$(wc -l "$decoratedFile.swp" | cut -f 1 -d ' ' )
+        if [ -s "$decoratedFile.swp" ] && [ "$decoratedFileLength" -gt 1 ]; then
+            mv $decoratedFile.swp $decoratedFile
+        else
+            echo "ERROR: decorate_rnaseq_file baseline for {wildcards.accession} and {wildcards.metric}"
+            exit 1
+        fi
+        """
+
+
+rule decorate_transcripts_baseline:
+    """
+    Decorate rna-seq baseline transcripts with transcript name from the latest Ensembl release.
+    """
+    container: "docker://quay.io/ebigxa/ensembl-update-env:amm1.1.2"
+    log: "logs/{accession}-decorate_transcripts_baseline_{metric}.log"
+    resources:
+        mem_mb=get_mem_mb
+    input:
+        getagg=rules.summarize_transcripts.output
+    params:
+        organism=get_organism(),
+        transcripts="{accession}-transcripts-{metric}.tsv.undecorated.aggregated",
+        decotranscripts="{accession}-transcripts-{metric}.tsv"
+    output:
+        temp("logs/{accession}-transcripts-{metric}.tsv.done")
+    shell:
+        """
+        set -e # snakemake on the cluster doesn't stop on error when --keep-going is set
+        exec &> "{log}"
+        source {workflow.basedir}/bin/reprocessing_routines.sh
+
+        # transcripts undecorated-aggregated rule done
+        echo {input.getagg}
+
+        if [ -s {params.transcripts} ] ; then
+
+            geneNameFile=$( get_geneNameFile_given_organism {params.organism}  )
+            transcriptFile=$( get_transcriptFile_given_organism {params.organism}  )
+
+            echo $geneNameFile
+            echo $transcriptFile
+
+            test -s "$geneNameFile" || (  >&2 echo "$0 gene name file not found: $geneNameFile" ; exit 1 )
+            test -s "$transcriptFile" || (  >&2 echo "$0 transcript to gene mapping file not found: $transcriptFile" ; exit 1 )
+
+            decoratedFile={params.decotranscripts} 
+
+            # Ammonite REPL & Script-Runner
+            #export JAVA_OPTS="-Xmx3000M"
```

Comment:
thanks @pcm32  The flag `   export JAVA_OPTS="-Xmx3000M"` was used in the old setup. Actually only for baseline transcript decoration [here](https://github.com/ebi-gene-expression-group/atlas-prod/blob/develop/bioentity_annotations/decorate_routines.sh#L57). Ammonite man says the following: 

> Set a default heap limit of 500 megabytes to the amm standalone executable, so it no longer grows to the "default" of 1/4 your system memory (typically multiple gigabytes) unnecessarily. If you need to give it more memory, start it with JAVA_OPTS=-Xmx1500M amm

I don't think we have experienced problems in this rule, but I will provide ammonite available memory on the workflows (which depends on Snakemake attempts), just in case

Reply:
done 971c0d4